pkg/scaffold/api.go

```diff
@@ -269,12 +275,27 @@ func (api *API) scaffoldV2() error {
 	return nil
 }
 
-// Since we support single group only in v2 scaffolding, validate if resource
-// being created belongs to existing group.
-func (api *API) validateResourceGroup(r *resource.Resource) error {
+// isNewGroup will return true if the group informed is diff of the pre-existents
+func (api *API) isNewGroup(r *resource.Resource) bool {
 	for _, existingGroup := range api.project.ResourceGroups() {
-		if !strings.EqualFold(r.Group, existingGroup) {
-			return fmt.Errorf("group '%s' is not same as existing group '%s'. Multiple groups are not supported yet.", r.Group, existingGroup)
+		if strings.ToLower(r.Group) != strings.ToLower(existingGroup) {
+			return true
+		}
+	}
+	return false
+}
+
+// validateResourceGroup will return an error if the group cannot be created
+func (api *API) validateResourceGroup(r *resource.Resource) error {
+	if api.resourceExists() {
```

Comment:
You can not move the resource already existing check here, because if you use for example do a `kubebuilder create api --controller=false`, edit the type and then decide you want to create a controller, you would have to do `kubebuilder create api --resource=false` so that it only generates the controller this time, and if you do this check here you would not scaffold the controller.

Reply:
HI @Adirio, 

Note that the change here was just centralized the validations in the validate func. Where and when it is called was NOT changed at all. So, I mean I kept it as it is today since it is not part of the scope. 

See:

- If you run `kubebuilder create api --group webapp --version v1 --kind Guestbook`

> resource=false
> controller=true
**NOTE:** It will not work since the controller requires in the resource in the scaffold.  (it is a bug in the master which should not allow this option)

- if you run `kubebuilder create api --group webapp --version v1 --kind Guestbook`
> resource=true
> controller=false
OR both true it will works

- if run `kubebuilder create api --group webapp --version v1 --kind Guestbook`
> resource=true
> controller=false
And then, try again
With just controller true. 
Then, you will see the msg: `API resource already exists`
Which cames from:

```go
	if api.resourceExists() && !api.Force {
		return fmt.Errorf("API resource already exists")
	}
```

**And that was NOT changed at all.**

The current, behaviour is that is required to use the flag --force. 

Closing it as sort out.

Reply:
This is not resolved, unmark it please.

```bash
kubebuilder create api --group wepapp --version v1 --kind Guestbook --resource --controller=false
// Change things in kind
kubebuilder create api --group wepapp --version v1 --kind Guestbook --resource=false --controller --force
```

That is perfectly valid. The user thought he didnt need a controller and changed mind so added it later. The second time he doesn't want to remake the resource as he already added fields between the two command calls. Your PR is addding a bug as the validation is only done if `DoResource` is `true`, and as I show you above, you may want to create a controller and not a resource.

Reply:
Hi @Adirio, 

Please, check the explanation above. Note that the validation or when they are called was NOT changed at all. However, if you still checking any issue with, please could you provide the steps to reproduce the scenario which works in the master and not with the changes made in this PR?

Reply:
You are changing the order:

Consider `api.resourceExists()` returns `true`. In master the following steps are happening to scaffold a v2 resource.

1- `validateResourceGroup()` is called. It returns no error so we keep going. (lines 171-173)
2- The resource is scaffolded. (lines 175-213)
3- `api.resourceExists` is called to check if the resource should be added to the PROJECT file, and as it is `true`we do not add anything to the PROJECT file. (This was done to remove duplicates) (lines 215-223)
4- The controller is scaffolded if needed. (lines 233-256)
5- Main is updated. (lines 258-267)

But in your case:

1- `validateResourceGroup()` is called. It returns an error so we stop here. (lines 172-174)
2- The resource is **NOT** scaffolded. (lines 176-219)
3- The resource is not added to the PROJECT file. (lines 221-227)
4- The controller is **NOT** scaffolded if needed. (lines 237-264)
5- Main is **NOT** updated. (lines 266-275)

As you can see, in master the resource and the controller were scaffolded and main updated, while in your case it yields an error.

Now, the question may be, how do we reach line 171 (or 172 in your case) if `api.resourceExists()` returns `true`. And that has an easy answer, by using the `--force` flag.

1- Create an API: `kubebuilder create api --group webapp --version v1 --kind Guestbook --resource --controller=false`
2- Delete the generated files:
  - `api/v1/guestbook_types.go`
  - `api/v1/groupversion_info.go`
  - `config/samples/webapp_v1_guestbook.yaml`
  - `config/rbac/guestbook_editor_role.yaml`
  - `config/rbac/guestbook_viewer_role.yaml`
  - `config/crd/patches/webhook_in_guestbooks.yaml`
  - `config/crd/patches/cainjection_in_guestbooks.yaml`
  - `config/crd/kustomization.yaml`
  - `config/crd/kustomizeconfig.yaml`
3- Recreate the API: `kubebuilder create api --group webapp --version v1 --kind Guestbook --resource --controller=false --force`

To recreate the api we need to pass `--force` as the API is already in the PROJECT file. Master will recreate the API successfully while your case will not.

Reply:
Summary "the scenario which was affected is when the resource files are deleted manually and not from the PROJECT file and the force is used to re-created them"

It is checking now the force in order to allow re-create the resource which should solve it and allow us to keep the validations encapsulated and centralized as it is now. Could you please check?

Reply:
I added a small comment where a check you removed previously is needed back. As we can run that part by using `--force` we have to prevent it writting to the PROJECT file as it already has it.

Reply:
Solved it sees the other comment. 
Closing this one since is fixed and it is very verbose which can make harder others review the PR.